### PR TITLE
perf: add scaffold logic for custom api python template

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/component/generator/apiSpec/helper.ts [hunk 1]
```diff
@@ -966,6 +1008,24 @@ async function updateCodeForCustomApi(
...
+        .replace(/{{pathUrl}}/g, item.pathUrl)
+        .replace(/{{method}}/g, item.method);
+      actionsCode.push(code);
+    }
+
+    const botFilePath = path.join(destinationPath, "src", "bot.py");
+    const botFileContent = (await fs.readFile(botFilePath)).toString();
+    const updateBotFileContent = botFileContent
+      .replace("{{OPENAPI_SPEC_PATH}}", openapiSpecFileName)
+      .replace("# Replace with action code", actionsCode.join("\n"));
```
Comment (SLdragon) [inline packages/fx-core/src/component/generator/apiSpec/helper.ts:1027]: How about using similar placeholder {{ACTION_CODE}}

Comment (KennethBWSong) [reply packages/fx-core/src/component/generator/apiSpec/helper.ts:1027]: This part follow same pattern as ts/js. We may update them together in another pr if needed.

File: packages/fx-core/src/component/generator/apiSpec/helper.ts [hunk 2]
```diff
@@ -918,6 +930,36 @@ app.ai.action("{{operationId}}", async (context: TurnContext, state: Application
   return "result";
 });
   `,
+  python: `
```
Comment (SLdragon) [inline packages/fx-core/src/component/generator/apiSpec/helper.ts:933]: This helper.ts file is too huge, in the future, we can move different generators(typeb/sme/ai-generator) to different files.

**Review by SLdragon — APPROVED** [review]

File: packages/fx-core/src/component/generator/apiSpec/helper.ts [hunk 3]
```diff
@@ -801,7 +801,11 @@ async function updatePromptForCustomApi(
   language: string,
   chatFolder: string
 ): Promise<void> {
-  if (language === ProgrammingLanguage.JS || language === ProgrammingLanguage.TS) {
+  if (
+    language === ProgrammingLanguage.JS ||
+    language === ProgrammingLanguage.TS ||
+    language === ProgrammingLanguage.PY
+  ) {
```
Comment (xzf0587) [inline packages/fx-core/src/component/generator/apiSpec/helper.ts:808]: It is better to use meaningful condition like 
```javascript
// maintain an array to support ApiSpec feature
SupportApiLanguages = [
ProgrammingLanguage.JS,
ProgrammingLanguage.TS
ProgrammingLanguage.PY
]; 
// use condition like
SupportApiLanguages.includes(langueage)
```

**Review by yuqizhou77 — APPROVED** [review]

File: packages/fx-core/src/component/generator/apiSpec/helper.ts (hunk 3 above)
Comment (KennethBWSong) [reply packages/fx-core/src/component/generator/apiSpec/helper.ts]: Updated.

**Review by xzf0587 — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Review by frankqianms — APPROVED** [review]

**Pull request merged**